src/main/java/com/code_intelligence/jazzer/mutation/combinator/MutatorCombinators.java

```diff
@@ -301,6 +315,18 @@ public void mutateInPlace(T reference, PseudoRandom prng) {
         }
       }
 
+      @Override
+      public void crossOverInPlace(T reference, T otherReference, PseudoRandom prng) {
+        // Always cross over within the state of the first reference.
+        // State changes are introduced by mutateInPlace.
+        int currentState = getState.applyAsInt(reference);
+        if (currentState == -1) {
+          // Intermediate states are not supported in cross over.
+          return;
+        }
+        mutators[currentState].crossOverInPlace(reference, otherReference, prng);
```

Comment:
`otherReference` may not be a valid object of state `currentState` here - I don't think this cast is safe.